pkg/swagger/generator/model.go

```diff
@@ -161,73 +155,96 @@ func makeGenDefinitionHierarchy(name, pkg, container string, schema spec.Schema,
 		GenSchema:    pg.GenSchema,
 		DependsOn:    pg.Dependencies,
 		ExtraSchemas: gatherExtraSchemas(pg.ExtraSchemas),
-		Imports:      imports,
+		Imports:      collectSortedImports(pg.GenSchema),
 	}, nil
 }
 
-type importOpts struct {
-	pkg  string
-	name string
+type importStmt struct {
+	ImportPath string
+	AsName     string
+	MustAsName bool
 }
 
-func findImports(sch *GenSchema, opts *importOpts) map[string]string {
-	imp := map[string]string{}
-	if sch.Name != opts.name {
-		if sch.Pkg != opts.pkg {
-			// not in the same pkg, import pkg
-			if sch.Pkg != "" && sch.Module != "" {
-				imp[sch.Module] = sch.Pkg
-			}
-			// on cross pkg import, add pkg alias to type
-			if sch.PkgAlias != "" {
-				sch.KclType = sch.PkgAlias + "." + sch.KclType
+func collectSortedImports(model GenSchema) []importStmt {
+	importMap := map[string]importStmt{}
+	collectImports(&model, model.Pkg, importMap)
+	sortedPkgPaths := make([]string, 0, len(importMap))
+	sortedImports := make([]importStmt, 0, len(importMap))
+	for k := range importMap {
+		sortedPkgPaths = append(sortedPkgPaths, k)
+	}
+	sort.Strings(sortedPkgPaths)
+	for _, k := range sortedPkgPaths {
+		sortedImports = append(sortedImports, importMap[k])
+	}
+	return sortedImports
+}
+
+func getImportAsName(imp map[string]importStmt, sch *GenSchema) string {
+	parts := strings.Split(sch.Pkg, ".")
+	asName := ""
+	for i := len(parts) - 1; i >= 0; i-- {
+		conflict := false
+		// when conflict with other import as name, the `import as` name will be "{parentPkgName}strings.Title({PkgAlias})"
+		asName = parts[i] + strings.ToTitle(asName)
+		for _, v := range imp {
+			if v.AsName == asName {
+				conflict = true
+				break
 			}
 		}
+		if !conflict {
+			return asName
+		}
 	}
+	return "kusionMangled" + strings.ToTitle(asName)
```

Comment:
1. Why use `"kusionMangled"`?
2. What would happen if there were multiple conflicts?

Reply:
1. use `import as` to resolve possible import conflicts like `import io.k8s.apimachinery.pkg.apis.meta.v1` and `import io.k8s.api.core.v1`
2. concat parent package name as the prefix to resolve conflicts: `metaV1` and `coreV1`; If conflicts remain, continue concat until <a>. no conflict <b>. no parent package exists
3. If all the parent package name is added as the prefix and conflicts remain, a `kusionMangled` prefix will be added to resolve it. The `kusionMangled` prefix is just a fallback.
4. After all these are processed, if conflicts remain(which should not happen for well-designed APIs), we'll print a warning message and live the conflict for users to properly resolve.